logger/logWriter.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 ThoughtWorks, Inc.
+
+// This file is part of Gauge.
+
+// Gauge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Gauge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Gauge.  If not, see <http://www.gnu.org/licenses/>.
+
+package logger
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+)
+
+// Writer reperesnts to a custom writer.
+// It intercents the log messages and redirects them to logger according the log level given in info
+type Writer struct {
+	loggerID            string
+	ShouldWriteToStdout bool
+	stream              int
+}
+
+// LogInfo repesents the log message structure for plugins
+type LogInfo struct {
+	LogLevel string `json:"logLevel"`
+	Message  string `json:"message"`
+}
+
+func (w Writer) Write(p []byte) (int, error) {
+	m := &LogInfo{}
+	err := json.Unmarshal(p, m)
+	if err != nil {
+		m.LogLevel = "info"
+		m.Message = string(p)
+	}
+	if w.stream > 0 {
+		m.Message = fmt.Sprintf("[runner: %d] %s", w.stream, m.Message)
+	}
+	switch m.LogLevel {
+	case "debug":
+		Debug(w.ShouldWriteToStdout, w.loggerID, m.Message)
+	case "info":
+		Info(w.ShouldWriteToStdout, w.loggerID, m.Message)
+	case "error":
+		Error(w.ShouldWriteToStdout, w.loggerID, m.Message)
+	case "warning":
+		Warning(w.ShouldWriteToStdout, w.loggerID, m.Message)
+	case "fatal":
+		Fatal(w.ShouldWriteToStdout, w.loggerID, m.Message)
```

Comment:
Fatal error on a plugin should not end the Gauge process too. This needs to be handled differently.

Reply:
How would you recommend handling it?

Reply:
We need to aggregate all the fatal errors from the plugins and print it at the end of execution or print them when Gauge's Fataf is used.
Will be tracked as separate issue #1426